lib/smf/smf.c

```diff
@@ -229,7 +221,8 @@ void smf_set_initial(struct smf_ctx *ctx, const struct smf_state *init_state)
 
 	internal->is_exit = false;
 	internal->terminate = false;
-	ctx->current = init_state;
+	internal->handled = false;
+	internal->new_state = false;	ctx->current = init_state;
```

Comment:
Formatting quirk.  These statements should be on separate lines.

Reply:
Oh no :( I'm not sure I'm enough of a Git to know how to amend an earlier commit. Can I fix it by amending the last commit?

Reply:
FYI, my workflow is to use "fixup" commits to make changes to a multi-commit stack.

1. Make the changes to the unstaged file
2. Run `git commit -a --fixup <commit_hash>`.  For example `git commit -a --fixup HEAD~2` to fixup the 3rd commit from the top.  Or use the direct hash shown by `git log --oneline`
3. Repeat steps 1&2 for additional changes that need to target different commits.
4. Run `git rebase -i --autosquash`